pkg/controller/jobframework/reconciler.go

```diff
@@ -574,6 +595,40 @@ func (r *JobReconciler) ReconcileGenericJob(ctx context.Context, req ctrl.Reques
 	return ctrl.Result{}, nil
 }
 
+func shouldHandleDeletionOfDeactivatedWorkload(afterDeactivatedByKueue *time.Duration, wl *kueue.Workload) bool {
+	return afterDeactivatedByKueue != nil && !workload.IsActive(wl) && workload.IsEvictedDueToDeactivationByKueue(wl)
+}
+
+func (r *JobReconciler) handleWorkloadAfterDeactivatedPolicy(ctx context.Context, job GenericJob, wl *kueue.Workload) (time.Duration, error) {
+	if !shouldHandleDeletionOfDeactivatedWorkload(r.workloadRetentionPolicy.AfterDeactivatedByKueue, wl) {
```

Comment:
```suggestion
func (r *JobReconciler) shouldHandleDeletionOfDeactivatedWorkload(wl *kueue.Workload) bool {
	return r.workloadRetentionPolicy.AfterDeactivatedByKueue != nil && !workload.IsActive(wl) && workload.IsEvictedDueToDeactivationByKueue(wl)
}

func (r *JobReconciler) handleWorkloadAfterDeactivatedPolicy(ctx context.Context, job GenericJob, wl *kueue.Workload) (time.Duration, error) {
	if !shouldHandleDeletionOfDeactivatedWorkload(wl) {
```

I think we can reduce the arguments to add `JobReconciler` receiver to `shouldHandleDeletionOfDeactivattedWorkload`.